activemq-compat/pom.xml

```diff
@@ -0,0 +1,95 @@
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.apache.activemq</groupId>
+    <artifactId>activemq-parent</artifactId>
+    <version>6.2.1-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>activemq-compat</artifactId>
+  <packaging>bundle</packaging>
+  <name>ActiveMQ :: Compat</name>
+  <description>The ActiveMQ Compatibility (MRJAR)</description>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.felix</groupId>
+        <artifactId>maven-bundle-plugin</artifactId>
+        <extensions>true</extensions>
+        <inherited>true</inherited>
+        <configuration>
+          <instructions>
+            <Import-Package>
+              !java.*,
+              !javax.naming*,
+              !javax.net*,
+              !org.apache.activemq*,
+              !com.google.errorprone.annotations,
+              !com.google.errorprone.annotations.concurrent,
+              com.thoughtworks.xstream.*;resolution:="optional",
+              javax.jmdns.*;resolution:="optional",
+              *
+            </Import-Package>
+            <Private-Package>
+              com.google.errorprone.annotations,
+              com.google.errorprone.annotations.concurrent
+            </Private-Package>
+            <Multi-Release>true</Multi-Release>
+            <_noee>true</_noee>
+          </instructions>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <id>jdk21-plus</id>
+      <activation>
+        <jdk>[21,)</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>java21-compile</id>
+                <phase>compile</phase>
+                <goals>
+                  <goal>compile</goal>
+                </goals>
+                <configuration>
+                  <release>21</release> <!-- Specific Java version for alternative classes -->
```

Comment:
I probably wouldnt change anything for 21 users personally...but if going with 21 for this, then you may want to avoid, or split out, the rest of the shim beyond the Subject bit since if the methods are ever used the current approach would break anyone actually using the SecurityManager on 21-23 currently.